dask/dataframe/io/parquet/arrow.py

```diff
@@ -146,6 +150,73 @@ def _gather_metadata(
         parquet-to-ddf mapping (e.g. schema, partition_info).
     """
 
+    if use_pa_ds:
+        # Use pyarrow.dataset API
+
+        if len(paths) == 1 and fs.isdir(paths[0]):
+            paths = paths[0]
+
+        ds = pa_ds.dataset(
+            paths,
+            format="parquet",
+            partitioning=dataset_kwargs.get(
+                "partitioning", "hive"
+            ),  # Assume "hive" by default
+        )
+        schema = ds.schema
+        metadata = []
+        base = ""
+        # Dataset API doesn't use partition_info
+        partition_info = {
+            "partitions": None,
+            "partition_keys": {},
+            "partition_names": [],
+        }
+        if gather_statistics is None:
+            gather_statistics = True
+        if split_row_groups is None and gather_statistics:
+            split_row_groups = True
+
+        # TODO: Translate multiple/nested filters from `filters`.
+        #       For now, user must include the filters in `dataset_kwargs`
+        #       for "full" filtering support.
+        ds_filters = dataset_kwargs.get("filters", None)
+        if filters and ds_filters is None:
+            if not isinstance(filters, list):
+                raise ValueError("Outer-most type of filters must be a list.")
+            if len(filters) > 1 or not isinstance(filters[0], tuple):
+                warnings.warn(
+                    "Failed to translate filter for dataset API. "
+                    "Include filters based on pyarrow.dataset.Expression "
+                    "in dataset kwargs for full support."
+                )
+            else:
+                filter_col = filters[0][0]
+                filter_op = filters[0][1]
+                filter_op = {
+                    "==": operator.eq,
+                    "!=": operator.ne,
+                    ">=": operator.ge,
+                    "<=": operator.le,
+                    ">": operator.gt,
+                    "<": operator.lt,
+                }[filter_op]
+                filter_comp = filters[0][2]
+                ds_filters = filter_op(pa_ds.field(filter_col), filter_comp)
+
+        for file_frag in ds.get_fragments(filter=ds_filters):
+            for rg_frag in file_frag.split_by_row_group():
```

Comment:
Need to specify the filter here again to ensure row groups are skipped if possible

Reply:
Just to clarify - I be passing `predicate=ds_filters` to `split_by_row_group()`?

Reply:
> Just to clarify - I be passing predicate=ds_filters to split_by_row_group()?

Yes indeed. But that reminds me, we should rename that to `filter=` to be consistent with `get_fragments`

Reply:
I noticed that this doesn't work when filtering on a partitioning column.  For example, the new `test_pa_dataset_partitioned` fails with:

```
dask/dataframe/io/parquet/arrow.py:149: in _get_ds_metadata
    for rg_frag in file_frag.split_by_row_group(predicate=ds_filters):
pyarrow/_dataset.pyx:955: in pyarrow._dataset.ParquetFileFragment.split_by_row_group
    ???
pyarrow/_dataset.pyx:579: in pyarrow._dataset._insert_implicit_casts
    ???
pyarrow/error.pxi:122: in pyarrow.lib.pyarrow_internal_check_status
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pyarrow.lib.ArrowInvalid: Field named 'b' not found or not unique in the schema.

pyarrow/error.pxi:84: ArrowInvalid
```

Reply:
Ah, that's something we fixed recently, but you need to pass the dataset schema (because the fragment's parquet file itself has a schema without the partition fields). So `split_by_row_group(ds_filters, schema=schema)`